custom/lib_portal/models/res_partner.py

```diff
@@ -0,0 +1,22 @@
+from odoo import models, fields
+
+class ResPartner(models.Model):
+    _inherit = 'res.partner'
+    
+    def _isMember(self):
+        """
+        this Function is used to check the active customer membership details
+        """
+        return True
+
+    date_of_birth = fields.Date(string='Date of Birth', 
+                                default=fields.Date.context_today, 
+                                required=True)
```

Comment:
this field shouldn't be required because 'res.partner' is used for other purposes than the customers. For example, a company from where you buy products is a partner, and in that case, there is no date of birth.

Reply:
I agree..Thanks for pointing it out.Updated the code now